packages/react/src/date-picker/DatePicker.test.tsx

```diff
@@ -281,12 +257,23 @@ describe('DatePicker', () => {
 			dateFormat: 'd MMM yyyy',
 		});
 
+		const user = userEvent.setup();
+
 		// Type a valid date in the input field that isn't in the display format
-		await userEvent.type(await getInput(), '05 06 2023');
-		await userEvent.keyboard('{Tab}');
+		await user.type(
+			screen.getByRole('textbox', {
+				name: 'Example (e.g. 5 Aug 2015) (optional)',
+			}),
+			'05 06 2023'
+		);
+		await act(() => user.keyboard('{Tab}'));
```

Comment:
Most tests are fixed by these three lines:
1. Calling `setup` on `userEvent` and then using the single user instance returned rather the `userEvent` object directly.
2. Migrating away from the custom async `getInput` (which doesn't need to be async as it uses the sync `get...` method) and to a more specific version of that method that targets the accessible name.
3. Wrapping the `Tab` press in `act` as that's the state update step.